### PR TITLE
Fix bug Service Bus binder consumer cannot be automatically created Topic/Subscriptions

### PR DESCRIPTION
**Pull request by hui1110**
# Description
 In the current version, the Azure Service Bus binder cannot create topics/subscriptions from the consumer itself.
Fix issue: https://github.com/Azure/azure-sdk-for-java/issues/30324

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by saragluna** [comment]
Please update your PR description.

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/spring-cloud-azure-resourcemanager/src/test/java/com/azure/spring/cloud/resourcemanager/implementation/crud/ServiceBusTopicSubscriptionCrudTests.java [hunk 1]
```diff
@@ -80,4 +82,42 @@ Tuple3<String, String, String> getKey() {
...
+        ServiceBusNamespace namespace = mock(ServiceBusNamespace.class);
+
+        when(resourceManager.serviceBusNamespaces()).thenReturn(namespaces);
+        when(namespaces.getByResourceGroup(resourceMetadata.getResourceGroup(), NAMESPACE))
+            .thenReturn(namespace);
+
+        Topics topics = mock(Topics.class);
+        when(namespace.topics()).thenReturn(topics);
+        Assert.isNull(namespace.topics().getByName(TOPIC_NAME));
+
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-resourcemanager/src/test/java/com/azure/spring/cloud/resourcemanager/implementation/crud/ServiceBusTopicSubscriptionCrudTests.java:97]: nit: remove the empty line

File: sdk/spring/spring-cloud-azure-resourcemanager/src/test/java/com/azure/spring/cloud/resourcemanager/implementation/crud/ServiceBusTopicSubscriptionCrudTests.java [hunk 2]
```diff
@@ -80,4 +82,42 @@ Tuple3<String, String, String> getKey() {
...
+        when(namespace.topics()).thenReturn(topics);
+        Assert.isNull(namespace.topics().getByName(TOPIC_NAME));
+        when(topics.getByName(TOPIC_NAME)).thenReturn(topic);
+
+        Topic.DefinitionStages.Blank define = mock(Topic.DefinitionStages.Blank.class);
+        when(topics.define(TOPIC_NAME)).thenReturn(define);
+
+        Topic.DefinitionStages.WithCreate create = mock(Topic.DefinitionStages.WithCreate.class);
+        when(define.withNewSubscription(SUBSCRIPTION_NAME)).thenReturn(create);
+        when(create.create()).thenReturn(topic);
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-resourcemanager/src/test/java/com/azure/spring/cloud/resourcemanager/implementation/crud/ServiceBusTopicSubscriptionCrudTests.java:120]: Should we add an assertion here to make sure the topic & sub are created?

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/spring-cloud-azure-resourcemanager/src/test/java/com/azure/spring/cloud/resourcemanager/implementation/crud/ServiceBusTopicSubscriptionCrudTests.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    void topicDoesNotExistReturnNull() {
+        ServiceBusTopicCrud topicCrud = mock(ServiceBusTopicCrud.class);
+        ServiceBusTopicSubscriptionCrud topicSubCrud = new ServiceBusTopicSubscriptionCrud(this.resourceManager,
+            this.resourceMetadata, topicCrud);
+
+        when(topicCrud.get(Tuples.of(NAMESPACE, TOPIC_NAME))).thenReturn(null);
+
+        ServiceBusSubscription actualGet = topicSubCrud.get(getKey());
+        Assertions.assertNull(actualGet);
+    }
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-resourcemanager/src/test/java/com/azure/spring/cloud/resourcemanager/implementation/crud/ServiceBusTopicSubscriptionCrudTests.java:95]: Do we need a case for when topic is not null?

File: sdk/spring/spring-cloud-azure-resourcemanager/src/test/java/com/azure/spring/cloud/resourcemanager/implementation/crud/ServiceBusTopicSubscriptionCrudTests.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        when(namespaces.getByResourceGroup(resourceMetadata.getResourceGroup(), getKey().getT1()))
+            .thenReturn(namespace);
+
+        Topics topics = mock(Topics.class);
+        Topic topic = mock(Topic.class);
+        ServiceBusSubscriptions serviceBusSubscriptions = mock(ServiceBusSubscriptions.class);
+        when(namespace.topics()).thenReturn(topics);
+        when(topics.getByName(getKey().getT2())).thenReturn(topic);
+        when(topic.subscriptions()).thenReturn(serviceBusSubscriptions);
+        when(serviceBusSubscriptions.getByName(getKey().getT3())).thenThrow(exception);
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-resourcemanager/src/test/java/com/azure/spring/cloud/resourcemanager/implementation/crud/ServiceBusTopicSubscriptionCrudTests.java:67]: Seems like we can remove the namespaces in this case?

Try to minimize the unnecessary code.

File: sdk/spring/spring-cloud-azure-resourcemanager/src/test/java/com/azure/spring/cloud/resourcemanager/implementation/crud/ServiceBusTopicSubscriptionCrudTests.java [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+        when(namespaces.getByResourceGroup(resourceMetadata.getResourceGroup(), getKey().getT1()))
+            .thenReturn(namespace);
+
+        Topics topics = mock(Topics.class);
+        Topic topic = mock(Topic.class);
+        ServiceBusSubscriptions serviceBusSubscriptions = mock(ServiceBusSubscriptions.class);
+        when(namespace.topics()).thenReturn(topics);
+        when(topics.getByName(getKey().getT2())).thenReturn(topic);
+        when(topic.subscriptions()).thenReturn(serviceBusSubscriptions);
+        when(serviceBusSubscriptions.getByName(getKey().getT2())).thenThrow(exception);
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-resourcemanager/src/test/java/com/azure/spring/cloud/resourcemanager/implementation/crud/ServiceBusTopicSubscriptionCrudTests.java:49]: If we can mock the `ServiceBusTopicCrud`, we can reduce the code here?

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/spring-cloud-azure-resourcemanager/src/test/java/com/azure/spring/cloud/resourcemanager/implementation/crud/ServiceBusTopicSubscriptionCrudTests.java (hunk 3 above)
Comment (hui1110) [reply sdk/spring/spring-cloud-azure-resourcemanager/src/test/java/com/azure/spring/cloud/resourcemanager/implementation/crud/ServiceBusTopicSubscriptionCrudTests.java:89]: added

**Review by saragluna — APPROVED** [review]

**Pull request merged**